Source/API/RequestBuilder.swift

```diff
@@ -7,50 +7,85 @@
 
 import UIKit
 
-class RequestBuilder {
+final class RequestBuilder {
 
-    private let requestParameters: RequestParameters
+    private let configuration: ClientConfiguration
+    private let authScheme = "OMGClient"
 
-    init(requestParameters: RequestParameters) {
-        self.requestParameters = requestParameters
+    init(configuration: ClientConfiguration) {
+        self.configuration = configuration
     }
 
     func buildHTTPURLRequest(withEndpoint endpoint: APIEndpoint) throws -> URLRequest {
-        guard let requestURL = endpoint.makeURL(withBaseURL: self.requestParameters.baseURL()) else {
-            throw OMGError.configuration(message: "Invalid request")
+        guard let requestURL = URL(string: configuration.baseURL)?.appendingPathComponent(endpoint.path) else {
+            throw OMGError.configuration(message: "Invalid base url")
         }
+
         var request = URLRequest(url: requestURL)
         request.httpMethod = "POST"
         request.cachePolicy = .useProtocolCachePolicy
         request.timeoutInterval = 6.0
-        try self.addRequiredHeaders(toRequest: &request)
-        endpoint.additionalHeaders?.forEach({ (key, value) in
-            request.addValue(value, forHTTPHeaderField: key)
-        })
+
+        try addRequiredHeaders(toRequest: &request)
+
+        // Add extra header when endpoint is transactionRequestConsume.
```

Comment:
Can we move this somehow in a different function, this may grow in the future if we have more endpoints that require to set the idempotency token. Maybe this requires a deeper refactoring.

Reply:
Does the idempotency token change over time, or based on the endpoint? if not, we might want to remove it from request parameter.

Reply:
It's just a param that you need to add to the request when consuming a consumption, but instead of going to the request parameters, it goes to the headers

Reply:
Instead of moving this logic in a different function, I created `APIParameters` protocol (replacing `Parameterable`) to specify if the http task requires idempotency token or not, and add the new parameter in `HTTPTask` (originally named `Task`) to get idempotency token from its parameters.

Reply:
there is one concern here that one might forget to add `getIdempotencyToken()` method to api parameters since there is a default implementation of it. but without default implementation, one needs to implement it every api parameters.